### PR TITLE
floorbots can pass through firelocks

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl: 
feature: floorbots can pass through firelocks
/:cl:

they are really useless, as they will just be stuck outside of the place that needs repairs, now they can pass under firelocks and be at same level of usefulness as medbots and cleanbots. their sprite layer  is under the doors too

### CODE REVIEW

**Comment by MMMiracles** [comment]
floorbot-sized doggy door on firelocks when

**Comment by Tlaltecuhtli** [comment]
thats the lore explaination

**Comment by ATH1909** [comment]
does their extinguisher work properly while they're in a firelock? what if their extinguisher is in emagged mode?

**Comment by Farquaar** [comment]
Subtle, but clever change.

**Comment by ATH1909** [comment]
wait, oh, this PR is for floorbots.

can firebots be given this flag as well, since they're supposed to, y'know, put out fires?

**Comment by cacogen** [comment]
Now understand I say this as usual not expecting it to do anything but describing the doggy door is just like what the fuck it's easier to overlook it without that justification of something that doesn't make sense and isn't visible on the sprite anyway

**Comment by LemonInTheDark** [comment]
I don't like this, it seems contrived and kinda silly

File: code/modules/projectiles/projectile/bullets/rifle.dm [hunk 1]
```diff
@@ -11,7 +11,7 @@
 	icon_state = "gaussphase"
 	damage = 20
 	armour_penetration = 70
-	pass_flags = PASSTABLE | PASSGLASS | PASSGRILLE | PASSCLOSEDTURF | PASSMACHINE | PASSSTRUCTURE | PASSDOORS
+	pass_flags = PASSTABLE | PASSGLASS | PASSGRILLE | PASSCLOSEDTURF | PASSMACHINE | PASSSTRUCTURE | PASSDOORS | PASSFIRELOCK
```
Comment (Mothblocks) [inline code/modules/projectiles/projectile/bullets/rifle.dm:14]: what the fuck?

**Comment by Mothblocks** [comment]
i also think this is contrived and silly

**Pull request closed without merging**